src/Diagnostics.RuntimeHost/Controllers/Site/SiteControllerBase.cs

```diff
@@ -21,7 +21,7 @@ public SiteControllerBase(IServiceProvider services, IRuntimeContext<App> runtim
             this._siteService = (ISiteService)services.GetService(typeof(ISiteService));
         }
 
-        protected async Task<App> GetAppResource(string subscriptionId, string resourceGroup, string appName, DiagnosticSiteData postBody, DateTime startTime, DateTime endTime)
+        protected async Task<App> GetAppResource(string subscriptionId, string resourceGroup, string appName, DiagnosticSiteData postBody, DateTime startTime, DateTime endTime, bool fetchStackInformation = true)
```

Comment:
Should we put the default value as false Shekhar?

Reply:
the issue is if we put it to false, then we would have to update lot of places. So the idea here is maintain the current behavior of this method, and only override the behavior at certain places where needed

Reply:
GetDetector and other places are still going to fetch the Tenant information. The total calls will be a lot less than it is right now but for an analysis / parent child there will still be concurrent calls for tenant id. What would break if we return SiteStack.All for all of them ? Do you think detectors authors are using the stack filter attribute to define the applicability of their detectors ?

Reply:
I guess you are talking about Stack and not Tenant. This PR is only taking care of fetching Stacks.
As I explained earlier also, the problem in removing this call from GetDetector currently is context will have Stack as StackType.All. So Detectors which are using cxt.Stack in their code will start to have issues of not showing stack specific data as they are doing right now.

The best way to attack the second problem is first we create a GIST that gets Stack Info, moves all the detectors using stack from Context to this GIST and then remove it from  our backend.